rust/cubetranspilers/src/error_reporter.rs

```diff
@@ -0,0 +1,45 @@
+use std::sync::{Arc, Mutex};
+
+use swc_core::common::errors::{DiagnosticBuilder, Emitter};
+
+pub struct ErrorReporter {
+    pub errors: Arc<Mutex<Vec<String>>>,
```

Comment:
Do we need this `Arc<Mutex>`? I assume you can borrow `&mut` over `self`.  Why? Because you don't run transpilers in parallel. It goes sequentially.

Then you can do whatever you want. `std::mem::replace` to reset the state after transpilation and return errors or `errors: Option<Vec<String>>` and then `self.errors.take()`. WDYT?

Another question: what do you need, Arc? 

`ErrorReporter` is created only once and goes into `Handler`.

Reply:
Yes, we need it because SWC requires thread safe types under the hood.